packages/runtime/container-runtime/src/summary/summaryFormats.md

```diff
@@ -0,0 +1,150 @@
+	| SummaryType.Attachment
+	| SummaryType.Blob
+	| SummaryType.Handle
+	| SummaryType.Tree;
+export type SummaryObject = ISummaryTree | ISummaryBlob | ISummaryHandle | ISummaryAttachment;
+```
+
+Each tree node in a summary tree is represented by the `ISummaryTree` interface above and contains the following:
+
+-   `type` - `SummaryType.Tree`
+-   `tree` - A list of children of the tree node. Each child node has the following format:
+
+    `<tree name>`: `<SummaryObject>` where `tree name` is the name given to the child node and `SummaryObject` is one of the following:
+
+    -   `ISummaryTree` - A subtree which has the same structure as this node. For example, container runtime will have subtrees for its data store which in turn will have subtrees for its DDSes.
+
+    -   `ISummaryBlob` - A blob that has some content for this node. For example, a DDS's node has a ".attributes" blob for its attributes such as type, snapshot versions, etc. It may have additional blobs for the actual user content in the DDS.
+
+        ```typescript
+        export interface ISummaryBlob {
+        	type: SummaryType.Blob;
+        	content: string | Uint8Array;
+        }
+        ```
+
+    -   `ISummaryAttachment` - This is used for attachment blobs that are uploaded to and downloaded from storage directly. The contents of these blobs are not part of the summary. Storage returns an ID after the attachment has been uploaded and this ID is included in the summary attachment. For example, a DDS can upload some of its large content directly to storage and include its ID in the summary.
+
+        ```typescript
+        export interface ISummaryAttachment {
+        	type: SummaryType.Attachment;
+        	id: string;
+        }
+        ```
+
+    -   `ISummaryHandle` - This is used for incremental summaries. If a node hasn't changed since the last successful summary, it can send a summary handle with a path to the summary tree object in the last summary. For example, if a data store or DDS did not change since the last summary, the runtime will use a handle for the entire data store instead of re-sending the entire subtree. The same concept applies for a blob or an attachment that did not change. An example of handle path that a DDS sends would be: "/_dataStoreId_/_ddsId_".
+
+        ```typescript
+        export interface ISummaryHandle {
+        	type: SummaryType.Handle;
+        	handleType: SummaryTypeNoHandle;
+        	handle: string;
+        }
+        export type SummaryTypeNoHandle =
+        	| SummaryType.Tree
+        	| SummaryType.Blob
+        	| SummaryType.Attachment;
+        ```
+
+## Snapshot Format
+
+Snapshot is downloaded from the storage and is the container's state at a point in time. A snapshot is represented by an `ISnapshotTree` which is defined in [this file](../../../../../common/lib/protocol-definitions/src/storage.ts) and has the following basic structure (some optional properties are removed for simplicity):
```

Comment:
A snapshot is a "JSON" (I think?) representation of the complete state of a container at a certain point of time. It is downloaded when ___. We represent a snapshot by `ISnapshotTree` defined [here](../../../../../common/lib/protocol-definitions/src/storage.ts), which looks something like:

Reply:
> A snapshot is a "JSON" (I think?) representation of the complete state of a container at a certain point of time.

I would not say "JSON" because that's implementation details. A server may choose to send the snapshot as a JSON but that's not what the runtime expects. It expects an `ISnapshotTree` format.

> It is downloaded when ___.

I added `typically on document load`. I wouldn't go into too much detail about snapshot and when it's used, etc. since those could change and other documents should explain that. This is just about describing the formats.